components/cluster-coordinator/org.wso2.carbon.cluster.coordinator/org.wso2.carbon.cluster.coordinator.rdbms/src/main/java/org/wso2/carbon/cluster/coordinator/rdbms/RDBMSCoordinationStrategy.java

```diff
@@ -283,25 +317,50 @@ private CoordinatorElectionTask(String nodeId, String groupId, Map<String, Objec
             this.currentNodeState = NodeState.MEMBER;
         }
 
-        @Override
-        public void run() {
+        public void runCoordinationElectionTask (long currentHeartbeatTime) {
             try {
                 if (log.isDebugEnabled()) {
                     log.debug("Current node state: " + currentNodeState);
                 }
+                long timeTakenForMemberTasks[] = new long[4];
+                long timeTakenForCoordinatorTasks[] = new long[5];
                 switch (currentNodeState) {
                     case MEMBER:
-                        performMemberTask();
+                        performMemberTask(currentHeartbeatTime, timeTakenForMemberTasks);
                         break;
                     case COORDINATOR:
-                        performCoordinatorTask();
+                        performCoordinatorTask(currentHeartbeatTime, timeTakenForCoordinatorTasks);
                         break;
                 }
+                long clusterTaskEndingTime = System.currentTimeMillis();
+                if (log.isDebugEnabled() && clusterTaskEndingTime - currentHeartbeatTime > 1000) {
```

Comment:
Shall we make this as a warning log for custerTaskEndTime-currentHeartbeatTime>heartbeatIntercal

Reply:
@suhothayan mostly the heartbeat is updated within few milliseconds. If we keep this at a low level, we will be able to find react before it increases up to an alerting level.
+1 to make it as warn logs

Reply:
Keep it as debug logs as well.